apps/main/src/dex/components/PagePool/Deposit/components/FormStake.tsx

```diff
@@ -149,25 +164,31 @@ const FormStake = ({ curve, poolData, poolDataCacheOrApi, routerParams, seed, us
   // steps
   useEffect(() => {
     if (curve && poolId) {
-      const updatedSteps = getSteps(activeKey, curve, poolData, formValues, formStatus, steps)
+      const updatedSteps = getSteps(activeKey, config, curve, poolData, formValues, formStatus, steps)
       setSteps(updatedSteps)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [chainId, poolId, signerAddress, formValues, formStatus])
+  }, [config, chainId, poolId, signerAddress, formValues, formStatus])
 
   const activeStep = signerAddress ? getActiveStep(steps) : null
   const disableForm = seed.isSeed === null || formStatus.formProcessing
-  const balLpToken = (userPoolBalances?.lpToken as string) ?? '0'
+
+  const { address: userAddress } = useConnection()
+  const { lpTokenBalance, isLoading: lpTokenBalanceLoading } = usePoolTokenDepositBalances({
+    chainId,
+    userAddress,
+    poolId,
+  })
 
   return (
     <>
       {/* input fields */}
       <FieldsWrapper>
         <FieldLpToken
           amount={formValues.lpToken}
-          balance={balLpToken}
-          balanceLoading={balancesLoading}
-          hasError={haveSigner ? new BigNumber(formValues.lpToken).isGreaterThan(balLpToken as string) : false}
+          balance={lpTokenBalance ?? '0'}
```

Comment:
I know it was there already, but omg it looks so bad when your balance is 0 during loading.